code/game/machinery/seed_extractor.dm

```diff
@@ -8,16 +8,53 @@
 
 /obj/structure/machinery/seed_extractor/attackby(obj/item/O as obj, mob/user as mob)
 
+	// Plant bag and other storage containers.
+	if(istype(O,/obj/item/storage))
+		var/obj/item/storage/container = O
+		if(length(container.contents) > 0)
```

Comment:
vero got a point, this code nests with if statements that could be early returned pretty deep, you can early return this. Essentially make the if statement opposite and then return in it with a message that you failed for whatever reason. 

so essenially this 

```dm
if(!correct)
     to_world("you failed")
     return
code here
```
instead of
```dm
if(correct)
   code here
else
    to_world("you failed")
```
or in this case swap the > with = or wrap them in brackets() and put ! infront of them to flip the result. 

`if(!(result == 0))`